### ISSUE TITLE
Unable to copy to FTP Write-Only Directories

### ISSUE THREAD

**Issue by twentyNoise** [rclone/rclone#8079]
#### The associated forum post URL from `https://forum.rclone.org`

https://forum.rclone.org/t/ftp-write-only-directories/45599/10

#### What is the problem you are having with rclone?

Inability to write to "write only directories" via FTP

#### What is your rclone version (output from `rclone version`)

rclone v1.68.0
- os/version: darwin 13.6.1 (64 bit)
- os/kernel: 22.6.0 (x86_64)
- os/type: darwin
- os/arch: amd64
- go/version: go1.23.1
- go/linking: dynamic
- go/tags: cmount

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

MacOS Ventura 13.6.1 64 bit

#### Which cloud storage system are you using? (e.g. Google Drive)

FTP

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

`/usr/local/bin/rclone copy -vv --no-check-dest --no-traverse --ignore-checksum --ignore-size --no-update-modtime --no-update-dir-modtime --ignore-times --inplace --retries 1 --low-level-retries 1 --exclude ".DS_Store" --stats 3s --log-file="/Users/TN/Documents/rclone.txt" "/Users/TN/Desktop/Testing.txt" FTP:"__Dropbox"`

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
2024/09/16 10:16:11 DEBUG : rclone: Version "v1.68.0" starting with parameters ["/usr/local/bin/rclone" "copy" "-vv" "--no-check-dest" "--no-traverse" "--ignore-checksum" "--ignore-size" "--no-update-modtime" "--no-update-dir-modtime" "--ignore-times" "--inplace" "--retries" "1" "--low-level-retries" "1" "--stats" "3s" "--log-file=/Users/TN/Documents/rclone.txt" "/Users/TN/Desktop/Testing.txt" "FTP:__Dropbox"]
2024/09/16 10:16:11 DEBUG : Creating backend with remote "/Users/TN/Desktop/Testing.txt"
2024/09/16 10:16:11 DEBUG : Using config file from "/Users/TN/.config/rclone/rclone.conf"
2024/09/16 10:16:11 DEBUG : fs cache: adding new entry for parent of "/Users/TN/Desktop/Testing.txt", "/Users/TN/Desktop"
2024/09/16 10:16:11 DEBUG : Creating backend with remote "FTP:__Dropbox"
2024/09/16 10:16:11 DEBUG : ftp://ftp.*****.com:21/__Dropbox: Connecting to FTP server
2024/09/16 10:16:11 DEBUG : ftp://ftp.*****.com:21/__Dropbox: dial("tcp","ftp.*****.com:21")
2024/09/16 10:16:11 DEBUG : ftp://ftp.*****.com:21/__Dropbox: > dial: conn=*fshttp.timeoutConn, err=<nil>
2024/09/16 10:16:12 DEBUG : Testing.txt: Need to transfer - File not found at Destination
2024/09/16 10:16:12 DEBUG : ftp://ftp.*****.com:21/__Dropbox: dial("tcp","XXX.XXX.XXX.XXX:XXXXX")
2024/09/16 10:16:12 DEBUG : ftp://ftp.*****.com:21/__Dropbox: > dial: conn=*tls.Conn, err=<nil>
2024/09/16 10:16:12 ERROR : Testing.txt: Failed to copy: SetModTime: 550 File not found
2024/09/16 10:16:12 ERROR : Attempt 1/1 failed with 1 errors and: SetModTime: 550 File not found
2024/09/16 10:16:12 INFO  : 
Transferred:   	          0 B / 0 B, -, 0 B/s, ETA -
Errors:                 1 (retrying may help)
Elapsed time:         0.7s

2024/09/16 10:16:12 DEBUG : 6 go routines active
2024/09/16 10:16:12 DEBUG : ftp://ftp.*****.com:21/__Dropbox: closing 1 unused connections
2024/09/16 10:16:12 NOTICE: Failed to copy: SetModTime: 550 File not found
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
Does this flag help?

      --no-update-modtime                           Don't update destination modtime if files identical

**Comment by twentyNoise** [comment]
> Does this flag help?
> 
> ```
>   --no-update-modtime                           Don't update destination modtime if files identical
> ```

Hi,

Unfortunately it does not. This flag is actually included in my testing above, which is also why I thought it odd that I received the specific “SetModTime” error.

**Comment by ncw** [comment]
Please give this a try with the `--ftp-no-check-upload` flag

[v1.69.0-beta.8309.f6c8cbd9d.fix-8079-ftp-write-only-upload](https://beta.rclone.org/branch/fix-8079-ftp-write-only-upload/v1.69.0-beta.8309.f6c8cbd9d.fix-8079-ftp-write-only-upload/) on branch [fix-8079-ftp-write-only-upload](https://github.com/rclone/rclone/tree/fix-8079-ftp-write-only-upload) (uploaded in 15-30 mins)

If it doesn't work then please supply logs with `-vv`.

I haven't actually tested these changes as I can't find a server which allows anonymous upload any more and I'm too lazy to set one up!

You should be able to remove some of these flags too - I suspect `--ignore-checksum --ignore-size --no-update-modtime --no-update-dir-modtime --ignore-times` won't be needed. I think you will need `--no-check-dest --no-traverse --inplace`

It would be interesting if you could find the minimum set of flags needed so we can write that in the docs.

**Comment by twentyNoise** [comment]
@ncw this works!

I used only the `--ftp-no-check-upload` and the `--inplace` flags. I'm going to do some additional testing tomorrow and I may keep the `--no-check-dest --no-traverse` flags just in case.

Thank You so much! I have been unable to use rclone for the past year because of this small issue.

**Comment by ncw** [comment]
Thanks for testing.

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.69
